include/envoy/server/filter_config.h

```diff
@@ -288,7 +288,7 @@ class NamedHttpFilterConfigFactory {
    */
   virtual Router::RouteSpecificFilterConfigConstSharedPtr
   createRouteSpecificFilterConfig(const Protobuf::Message&, FactoryContext&) {
-    return nullptr;
+    return nullptr; // GCOVR_EXCL_LINE
```

Comment:
Weirdly, line 290 is tagged as not covered. What do you think of excluding all this code with a single GCOVR_EXCL_START/GCOVR_EXCL_STOP pair?

Reply:
(The same weirdness occurs several times in this file.)

Reply:
This strikes me as a little shady. If these functions aren't needed, why aren't they pure virtual? If they are needed why aren't they tested?

Reply:
I think there are two things here: createEmptyProto/createFilterFactoryFroProto are implemented by default to aid v1-to-v2 transition. Empirically all the implementations of this class in the repo override both of them, which is why they get no code coverage. That said, as part of deprecating v1 these should become pure virtual, which makes this exercise moot.

createEmptyRouteConfigProto/createRouteSpecificFilterConfig are for HTTP router-specific filter configs. They can be made pure virtual if we implement them in EmptyHttpFilterConfig (again possibly breaking filters implemented outside the repo). However, the implementations (in either location) don't get invoked during tests because these methods only get called if there's a cluster with a per-filter-config for a given filter name, and we don't test what happens if you specify a per-filter-configs for a filter that doesn't override these methods.

I'm inclined to leave these for now.

Reply:
> don't get invoked during tests because these methods only get called if there's a cluster with a per-filter-config for a given filter name, and we don't test what happens if you specify a per-filter-configs for a filter that doesn't override these methods.

Isn't this missing test coverage? I'm not saying you need to fix it but shouldn't it count as missing?

Reply:
Definitely missing coverage. (When I said "leave these," I meant not exclude them from coverage.)

Reply:
Ah ok SGTM.